src/jit/compiler.h

```diff
@@ -297,6 +297,208 @@ enum RefCountState
+//    basically a list of this class and a const_iterator of that
+//    list.
+//
+class VariableLiveRange
+{
+public:
+    emitLocation               m_StartEmitLocation; // first position from where "m_VarHome" becomes valid
+    emitLocation               m_EndEmitLocation;   // last position where "m_VarHome" is valid
+    CodeGenInterface::siVarLoc m_VarHome;           // variable home
+
+    VariableLiveRange(CodeGenInterface::siVarLoc varHome, emitLocation startEmitLocation, emitLocation endEmitLocation)
+        : m_StartEmitLocation(startEmitLocation), m_EndEmitLocation(endEmitLocation), m_VarHome(varHome)
+    {
+    }
+};
+
+typedef jitstd::list<VariableLiveRange> LiveRangeList;
+typedef LiveRangeList::const_iterator   LiveRangeListIterator;
+
+#if DEBUG
+// Dump "VariableLiveRange" when code has not been generated and we don't have so the assembly native offset
+// but at least "emitLocation"s and "siVarLoc"
+void dumpVariableLiveRange(const VariableLiveRange* varLiveRange, const CodeGenInterface* codeGen);
+
+// Dump "VariableLiveRange" when code has been generated and we have the assembly native offset of each "emitLocation"
+void dumpVariableLiveRange(const VariableLiveRange* varLiveRange, emitter* _emitter, const CodeGenInterface* codeGen);
+
+//--------------------------------------------
+//
+// LiveRangeDumper: Used for debugging purposes during code
+//  generation on genCodeForBBList. Keeps an iterator to the fist
+//  edited/added "VariableLiveRange" of a variable during the
+//  generation of code of one block.
+//
+// Notes:
+//  The first "VariableLiveRange" reported for a variable during
+//  a BasicBlock is sent to "setDumperStartAt" so we can dump all
+//  the "VariableLiveRange"s from that one.
+//  After we dump all the "VariableLiveRange"s we call "reset" with
+//  the "liveRangeList" to set the barrier to nullptr or the last
+//  "VariableLiveRange" if it is opened.
+//  If no "VariableLiveRange" was edited/added during block,
+//  the iterator points to the end of variable's LiveRangeList.
+//
+class LiveRangeDumper
+{
+private:
+    LiveRangeListIterator m_StartingLiveRange; // Iterator to the first edited/added position
+                                               // during actual block code generation. If last
+                                               // block had a closed "VariableLiveRange" (with
+                                               // a valid "m_EndEmitLocation") and not changes
+                                               // were applied to variable liveness, it points
+                                               // to the end of variable's LiveRangeList.
+    bool m_hasLiveRangestoDump;                // True if a live range for this variable has been
+                                               // reported from last call to EndBlock
+
+public:
+    LiveRangeDumper(const LiveRangeList* liveRanges);
+
+    ~LiveRangeDumper(){};
```

Comment:
Not only that it doesn't do anything but the JIT doesn't generally use destructors because there are no resources to be released. The arena allocator doesn't release memory and there's pretty much nothing else that needs release (e.g. opened files).

Reply:
How the arena allocator deals with the memory it no longer use?

Reply:
When a method compilation ends all the arena allocated memory is released. Works quite well if you look at a memory stats dump:
```
Distribution of total memory allocated per method (in KB):
     <=         64 ===>   16630 count ( 56% of total)
     65 ..     128 ===>    8713 count ( 86% of total)
    129 ..     192 ===>    2040 count ( 93% of total)
    193 ..     256 ===>     950 count ( 96% of total)
    257 ..     512 ===>     676 count ( 98% of total)
    513 ..    1024 ===>     228 count ( 99% of total)
   1025 ..    4096 ===>      65 count ( 99% of total)
   4097 ..    8192 ===>       1 count (100% of total)
```
As you can see, more than half of the methods in corelib need only 64 kbytes. And 99% need less than 1 megabyte. While piecemeal memory release (and of course, reuse of release memory) would certainly improve this, that also comes with a throughput cost - destructors, delete/free calls, maintaining freed memory lists etc.

Reply:
Okey, I will remove all the destructors.